ocaml/typing/typedtree.ml

```diff
@@ -18,6 +18,16 @@
 open Asttypes
 open Types
 
+type constant =
+    Const_int of int
+  | Const_char of char
+  | Const_string of string * Location.t * string option
+  | Const_float of string
+  | Const_unboxed_float of string
+  | Const_int32 of int32
+  | Const_int64 of int64
+  | Const_nativeint of nativeint
```

Comment:
Is it much churnier to prefix these names with a `T`? I'd prefer that as it would then distinguish these constructors from the constructors with the same names in `Asttypes`.

Reply:
I believe the idea is to eventually share this type with lambda and completely replace the usage of `Asttypes.constant`. In which case, I'm not sure if renaming it now and changing it back shortly after is worth the effort.